pandas/core/indexes/datetimelike.py

```diff
@@ -731,7 +741,7 @@ def _add_delta_tdi(self, other):
         if self.hasnans or other.hasnans:
             mask = (self._isnan) | (other._isnan)
             new_values[mask] = iNaT
-        return new_values.view(self.dtype)
+        return new_values.view('i8')
```

Comment:
leave this

Reply:
This fixes adding tz-aware `DatetimeIndex` and `TimedeltaIndex`, since `self.dtype` here is a `Datetime64TZDtype` which causes `new_values.view(..)` to raise an exception. Would you prefer another solution, say checking if `is_datetime64tz_dtype(self.dtype)` first?

Reply:
yes use is_datetime64tz_dtype(self.dtype) is the correct idiom. but you'll have to demonstrate this bug specifically

Reply:
Posted stack trace that hits this line above. So I can change this to
```python
if is_datetime64tz_dtype(self.dtype):
  return new_values.view('i8')
return new_values.view(self.dtype)
```

Reply:
this is the not the right fix.

tz-aware need to be turned into local i8, added, then turned back into the original timezone via localization. normally we *do* add i8 directly, BUT you cannot do this with tz-aware, e.g. the days are shifted otherwise (imagine adding days), or don't work when you cross DST.

pandas.tseries.offsets  already does all of this. I haven't stepped thru this code lately, but need to hit a similar path to that.